specs/Standards/MultiToken/README.md

```diff
@@ -0,0 +1,539 @@
+    /// function call. It allows you to attach any Fungible or Non Fungible Token in a call to a
+    /// receiver contract.
+    ///
+    /// Requirements:
+    /// * Caller of the method must attach a deposit of 1 yoctoⓃ for security
+    ///   purposes
+    /// * Contract MUST panic if called by someone other than token owner or,
+    ///   if using Approval Management, one of the approved accounts
+    /// * The receiving contract must implement `mt_on_transfer` according to the
+    ///   standard. If it does not, MultiToken contract's `mt_resolve_transfer` MUST deal
+    ///   with the resulting failed cross-contract call and roll back the transfer.
+    /// * Contract MUST implement the behavior described in `mt_resolve_transfer`
+    ///
+    /// Arguments:
+    /// * `receiver_id`: the valid NEAR account receiving the token.
+    /// * `token_id`: the token to send.
+    /// * `amount`: amount of tokens to transfer for token_id
+    /// * `memo` (optional): for use cases that may benefit from indexing or
+    ///    providing information for a transfer.
+    /// * `msg`: specifies information needed by the receiving contract in
+    ///    order to properly handle the transfer. Can indicate both a function to
+    ///    call and the parameters to pass to that function.
+    fn mt_transfer_call(
+        &mut self,
+        receiver_id: AccountId,
+        token_id: TokenId,
+        amount: U128,
+        memo: Option<String>,
+        msg: String,
+    ) -> PromiseOrValue<U128>;
+
+    /// Batch token transfer. Transfer a tokens given token_ids and amounts. The token ids can correspond to  
+    /// either Non-Fungible Tokens or Fungible Tokens or some combination of the two. The token ids
+    /// are used to segment the types on a per contract implementation basis.
+    ///
+    /// Requirements
+    /// * Caller of the method must attach a deposit of 1 yoctoⓃ for security purposes
+    /// * Contract MUST panic if called by someone other than token owner or,
+    ///   if using Approval Management, one of the approved accounts
+    /// * `approval_id` is for use with Approval Management,
+    ///   see https://nomicon.io/Standards/NonFungibleToken/ApprovalManagement.html
+    /// * If using Approval Management, contract MUST nullify approved accounts on
+    ///   successful transfer.
+    /// * TODO: needed? Both accounts must be registered with the contract for transfer to
+    ///   succeed. See see https://nomicon.io/Standards/StorageManagement.html
+    /// * The token_ids vec and amounts vec must be of equal length and equate to a 1-1 mapping
+    ///   between amount and id. In the event that they do not line up the call should fail
+    ///
+    /// Arguments:
+    /// * `receiver_id`: the valid NEAR account receiving the token
+    /// * `token_ids`: the tokens to transfer
+    /// * `amounts`: the amount of tokens to transfer for corresponding token_id
+    /// * `approval_ids`: expected approval ID. A number smaller than
```

Comment:
I see reference to `approval_ids` here, but it's not in the function signature?

Reply:
Historically, at one point approval_ids I had added to the spec, but opted to yank it out, we'll bring this up today so I'll leave this here for the discussion.

Reply:
So from our last conversation, approval management is back in the spec